aten/src/ATen/native/Fill.cpp

```diff
@@ -10,7 +10,26 @@ namespace native {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ fill ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+namespace {
+  template <typename scalar_t>
+  inline void fill_fast(Tensor& self, Scalar value_scalar) {
+    auto value = value_scalar.to<scalar_t>();
+    scalar_t * dptr = reinterpret_cast<scalar_t *>(self.data_ptr());
```

Comment:
Use of a `reinterpret_cast` here shouldn't be necessary; `data_ptr` gives you a `void*` so you can just static cast it.